### PR TITLE
Upload weekly autoban report to pastebin

### LINKED ISSUE #2391: HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
Sentry Issue: [BOT-3CR](https://sentry.io/organizations/python-discord/issues/3793707044/?referrer=github_integration)

```
HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In content: Must be 2000 or fewer in length.
  File "discord/ext/tasks/__init__.py", line 239, in _loop
    await self.coro(*args, **kwargs)
  File "bot/exts/filters/filter_lists.py", line 308, in weekly_autoban_report_task
    await self.send_weekly_autoban_report()
  File "bot/exts/filters/filter_lists.py", line 346, in send_weekly_autoban_report
    await channel.send("\n\n".join(lines))
  File "discord/abc.py", line 1538, in send
    data = await state.http.send_message(channel.id, params=params)
  File "discord/http.py", line 744, in request
    raise HTTPException(response, data)

Unhandled exception in internal background task "weekly_autoban_report_task".
```

### PR DESCRIPTION
**Pull request by ChrisLovering**
This fixes #2391 and solves BOT-3CR which was caused by the weekly report being longer than 2000 characters.

Instead of posting the entire report to Discord, it is now upload to the pastebin, and uploaded as a text file to the channel, so people can use what they prefer.

### CODE REVIEW

**Review by vivekashok1221 — DISMISSED** [review]
Code looks good to me, will test after a few hours.
There's a typo in the docstring of `send_weekly_autoban_report` method, btw.
>    If chanel is not specified, it is sent to #mod-meta.
line 316.

**Comment by mbaruh** [comment]
@ChrisLovering can you resolve the merge conflict? it'll be moving the changes to the equivalent function in the new filtering.py
https://github.com/python-discord/bot/blob/ea0b227801f19d14338e7e582d745ea250773370/bot/exts/filtering/filtering.py#L1396

**Comment by ChrisLovering** [comment]
> @ChrisLovering can you resolve the merge conflict? it'll be moving the changes to the equivalent function in the new filtering.py
> 
> https://github.com/python-discord/bot/blob/ea0b227801f19d14338e7e582d745ea250773370/bot/exts/filtering/filtering.py#L1396

Rebased, I've also simplified it quite a lot, it was far more complicated than it needed to be.

File: bot/exts/filtering/filtering.py [hunk 1]
```diff
@@ -1432,7 +1434,17 @@ async def send_weekly_auto_infraction_report(self, channel: discord.TextChannel
         if len(lines) == 1:
             lines.append("Nothing to show")
 
-        await channel.send("\n\n".join(lines))
+        report = "\n\n".join(lines)
+        if len(report) > MAX_PASTE_LENGTH:
```
Comment (wookie184) [inline bot/exts/filtering/filtering.py:1438]: `MAX_PASTE_LENGTH` is the maximum length that can be sent to the pastebin which isn't what we want here. Not sure if we have a constant anywhere for message length.

**Review by wookie184 — CHANGES_REQUESTED** [review]

File: bot/exts/filtering/filtering.py (hunk 1 above)
Comment (ChrisLovering) [reply bot/exts/filtering/filtering.py:1438]: 🤦 I'll sort it

File: bot/exts/filtering/filtering.py [hunk 2]
```diff
@@ -1432,9 +1434,20 @@ async def send_weekly_auto_infraction_report(self, channel: discord.TextChannel
...
+            await channel.send(report)
+        except discord.HTTPException as e:
+            if e.code == 50035:  # Content too long
+                report = discord.utils.remove_markdown(report)
+                paste_url = await send_to_paste_service(report, extension="txt")
+                file_buffer = io.StringIO(report)
+                await channel.send(
+                    f"**{lines[0]}**\n\n{paste_url}",
+                    file=discord.File(file_buffer, "last_weeks_autoban_filters.txt"),
+                )
```
Comment (mbaruh) [inline bot/exts/filtering/filtering.py:1448]: Doesn't this suppress any other reasons for HTTP exceptions? you'll end up not having a report and then logging that it was sent successfully.

File: bot/exts/filtering/filtering.py [hunk 3]
```diff
@@ -1432,9 +1434,20 @@ async def send_weekly_auto_infraction_report(self, channel: discord.TextChannel
...
             lines.append("Nothing to show")
 
-        await channel.send("\n\n".join(lines))
-        log.info("Successfully sent auto-infraction report.")
+        report = "\n\n".join(lines)
+        try:
+            await channel.send(report)
+        except discord.HTTPException as e:
+            if e.code == 50035:  # Content too long
+                report = discord.utils.remove_markdown(report)
```
Comment (mbaruh) [inline bot/exts/filtering/filtering.py:1442]: What is the concern behind removing markdown before uploading it?

File: bot/exts/filtering/filtering.py [hunk 4]
```diff
@@ -1432,9 +1434,20 @@ async def send_weekly_auto_infraction_report(self, channel: discord.TextChannel
...
+                paste_url = await send_to_paste_service(report, extension="txt")
```
Comment (mbaruh) [inline bot/exts/filtering/filtering.py:1443]: This can raise exceptions that should be handled.

**Review by mbaruh — CHANGES_REQUESTED** [review]

File: bot/exts/filtering/filtering.py (hunk 2 above)
Comment (ChrisLovering) [reply bot/exts/filtering/filtering.py]: True, I've made is so that these still get raised.

**Review by mbaruh — APPROVED** [review]

**Pull request merged**